src/Nethermind/Nethermind.AuRa/Validators/ContractValidator.cs

```diff
@@ -116,13 +117,16 @@ public override void PreProcess(Block block, ProcessingOptions options = Process
         {
             var isProducingBlock = options.IsProducingBlock();
             var isProcessingBlock = !isProducingBlock;
+            var initBlock = InitBlockNumber == block.Number;
+            var loadValidators = _validators == null || isProducingBlock;
             
-            if (_validators == null || isProducingBlock)
+            if (loadValidators)
             {
                 Validators = LoadValidatorsFromContract(block.Header);
+                if(_logger.IsInfo && !_isProducing && isProcessingBlock) _logger.Info($"{(initBlock ? "Initial" : "Current")} contract validators ({Validators.Length}): [{string.Join<Address>(", ", Validators)}].");
```

Comment:
do not mix if(logger.isLevel) with business logic -may easy lead to errors

Reply:
why do we have _isProducing and isProducingBlock? can we make them more easily distinguished?

Reply:
now we have a strange looking !isProducing && isProcessing and the first impression is that one of them is always true. not abvious that one is a condig for isMining and other for current run

Reply:
changing structure and doing a little renaming.